packages/account/src/providers/provider.ts

```diff
@@ -846,7 +850,7 @@ Supported fuel-core version: ${supportedVersion}.`
       const hasMissingOutputs =
         missingOutputVariables.length !== 0 || missingOutputContractIds.length !== 0;
 
-      if (hasMissingOutputs) {
+      if (hasMissingOutputs && isTransactionTypeScript(transactionRequest)) {
```

Comment:
There's a lot of repetitions of `isTransactionTypeScript(transactionRequest)` throughout this function, perhaps we could just set that as a constant.

Reply:
I think you may have resolved this by accident @danielbate because I'm noit seeing any changes here

Reply:
In `estimateTxDependencies` I can only see one call to `isTransactionTypeScript` and one for `isTransactionTypeCreate`? I can only see one function in this class where we could reduce the repetition in `getTransactionCost`, but we clone the request so still need to cast `ScriptTransactionRequest`, I'd rather just call the function twice? Let me know if you think I'm missing something.